reference/vane-apis/ames.md

```diff
@@ -0,0 +1,303 @@
+information to `+decode-packet` which deserializes the packet and giving that data and the origin of the
+packet to `+on-hear-packet`.
+
+There are multiple `+on-hear` arms in `ames.hoon`. Here we refer to
+`on-hear:event-core`, as that is the one called by a `%hear` `task`. The other ones are used
+primarily for ack and nack processing, or receiving message fragments.
+
+#### Accepts
+ 
+```hoon
+[=lane =blob]
+```
+
+`%hear` takes in a `blob`, which is essentially a large atom (around 1kB or less)
+that is the raw data of the message, and a `lane`, which is the origin of the
+message (typically an IP address).
+
+#### Returns
+
+`%hear` can trigger any number of possible returns...?
+
+
+### %heed
+
+A vane can pass Ames a `%heed` `task` to request Ames track a peer's
+responsiveness.  If our `%boon`s to it start backing up locally,
+Ames will `give` a `%clog` back to the requesting vane containing the
+unresponsive peer's Urbit address.
+
+Stop tracking a peer by sending Ames a
+`%jilt` `task`.
+
+
+#### Accepts
+
+```hoon
+=ship
+```
+
+The ship to be tracked.
+
+#### Returns
+
+The `+on-heed` arm returns `event-core` with `heeds` modified to include a
+`duct` that points to the input `ship`. (this isn't a `gift` so maybe I should
+not include this?)
+
+If the `ship` is indeed being unresponsive, as measured by backed up `%boon`s,
+Ames will `give` a `%clog` `gift` to the requesting vane containing the
+unresponsive peer's urbit address.
+
+
+### %hole
+
+`%hole` handles packet crash notification. `%hole` works much like `%hear`, in
+that it passes an incoming raw packet to `+decode-packet` to be deserialized,
+and then passes that data along with the source of the packet to
+`+on-hear-packet` along with a `?` set to `%.n` denoting that there is something wrong with
+the packet. 
+
+#### Accepts
+
+```hoon
+[=lane =blob]
+```
+`%hole` takes in a `blob`, which is essentially a large atom (around 1kB or less)
+that is the raw data of the message, and a `lane`, which is the origin of the
+message (typically an IP address).
+
+#### Returns
+
+Same confusion as with `%hear`.
+
+
+### %jilt
+
+`%jilt` stops tracking a potentially unresponsive peer that was previously being
+tracked as a result of the `%heed` `task`.
+
+There are two `+on-jilt` arms, this `task` utilizes `on-hear:event-core`.
+
+#### Accepts
+
+```hoon
+=ship
+```
+
+The `ship` we no longer wish to track.
+
+#### Returns
+
+`+on-jilt` returns `event-core` with the `duct` associated to `ship` removed from `heeds`, assuming it is
+there. Otherwise it returns `event-core` unchanged.
+
+Does `%jilt` return any gifts?
+
+
+### %plea
+
+`%plea` is the `task` used to send messages over Ames. It extends the
+`%pass`/`%give` semantics across the network. As such, it is the most
+fundamental `task` in Ames and the primary reason for its existence.
+
+Ames `pass`es a `%plea` `note` to another vane when it receives a message on a
```

Comment:
```suggestion
Ames also `pass`es a `%plea` `note` to another vane when it receives a message on a
```

One of the confusing things about ames is that `%plea` is both a task and a note, so best to mark the difference.  (Incidentally, I think it would be even more confusing if they had different names)